src/middleware/auth.ts

```diff
@@ -65,5 +65,31 @@ async function adminAuthImpl(req: Request, res: Response, next: NextFunction) {
   }
 }
 
+async function passwordResetAuthImpl(req: Request, res: Response, next: NextFunction) {
+  try {
+    const token = req.header('x-auth-token');
```

Comment:
Can we just put this as a query param? Like `/passwordReset?resetToken=xxxxx`

Reply:
This one may lead to unintended confusion because the header is usually used for authentication

Reply:
I think just change this portion can merge alr